changes/8782.outlier_detection.rst

```diff
@@ -0,0 +1,3 @@
+Decrease the amount of file I/O required to compute the median when in_memory is set to False.
+Fix a bug that caused intermediate files to conflict for different slits when a MultiSlitModel was processed.
+Add association IDs to intermediate filenames.
```

Comment:
Under what conditions is this not the case on main? Running outlier detection using main on an association containing nircam images I see intermediate files with associations.
```
jw03516001001_02101_00001_nrca1_a3001_outlier_i2d.fits
...
jw03516001001_02101_00001_nrca1_a3001_median.fits
jw03516001001_02101_00001_nrca1_a3001_blot.fits
...
```

Reply:
You're right, this is not correct.  I checked the spectroscopic modes just now.  This only occurs for SourceModelContainer input, which is why I wrote the comment.  But in that case there is a single MultiSlitModel input to the step as opposed to an association, and we are looping over individual slits within that, so I think it's sensible that there would be no asn ID.